hta/utils/utils.py

```diff
@@ -174,3 +178,21 @@ def flatten_column_names(df: pd.DataFrame) -> None:
     """Flatten a DataFrame's a multiple index column names to a single string"""
     if isinstance(df.columns, pd.MultiIndex):
         df.columns = ["_".join(col).rstrip("_") for col in df.columns]
+
+
+def get_mp_pool_size(obj: object, num_objs: int) -> int:
+    """
+    Estimate the maximum pool size for multiprocessing
+
+    Args:
+        obj (object): one instance of the objects to be processed
+        num_objs (int): the total number of objects to be processed
+
+    Return:
+        the recommend pool size
+    """
+    obj_mem = get_memory_usage(obj)
```

Comment:
this is a recursive call on the json file and it is pretty slow. Is there a way to avoid doing this? It will eat away some of the speed up that multiprocessing will provide.

Reply:
I updated the way to estimate memory usage, it's less  time consuming(~10%) on my devserver.